### PR TITLE
[Add]: Login Registration setup with backend integration

### PR DESCRIPTION
**Pull request by nishihere19**
Signed-off-by: Nishi Agrawal <nishihere19@gmail.com>

### CODE REVIEW

File: webpack.config.ts [hunk 1]
```diff
@@ -191,7 +199,11 @@ const config: Configuration = {
 			systemvars: true, // load all the predefined 'process.env' variables which will trump anything local per dotenv specs.
 			silent: true, // hide any errors
 			defaults: false // load '.env.defaults' as the default values if empty.
-		})
+		}),
+		new Webpack.ProvidePlugin({
+			$: "jquery",
+			jQuery: "jquery"
+	})
```
Comment (AananthV) [inline webpack.config.ts:206]: Where are you using jquery?

File: src/routes/Registrationdetails.svelte [hunk 2]
```diff
@@ -2,12 +2,214 @@
   main {
```
Comment (AananthV) [inline src/routes/Registrationdetails.svelte:2]: Rename filename to CamelCase

File: src/routes/Registrationdetails.svelte [hunk 3]
```diff
@@ -2,12 +2,214 @@
...
+  function handlechange(e) {
+    let val = e.target.name;
+    if (val == 'name') state.name = e.target.value;
+    else if (val == 'department') state.department = e.target.value;
+    else if (val == 'batch') state.batch = e.target.value;
+    else if (val == 'phone') state.phone = e.target.value;
+    else if (val == 'email') state.email = e.target.value;
+    else if (val == 'password') state.password = e.target.value;
+    else state.confirmpass = e.target.value;
+    console.log(phoneInput.getNumber());
```
Comment (AananthV) [inline src/routes/Registrationdetails.svelte:43]: Remove console.log

File: src/routes/Registrationdetails.svelte [hunk 4]
```diff
@@ -2,12 +2,214 @@
...
+  };
+  function handlechange(e) {
+    let val = e.target.name;
+    if (val == 'name') state.name = e.target.value;
+    else if (val == 'department') state.department = e.target.value;
+    else if (val == 'batch') state.batch = e.target.value;
+    else if (val == 'phone') state.phone = e.target.value;
+    else if (val == 'email') state.email = e.target.value;
+    else if (val == 'password') state.password = e.target.value;
+    else state.confirmpass = e.target.value;
```
Comment (AananthV) [inline src/routes/Registrationdetails.svelte:42]: `state[val] = e.target.value` ?

File: src/routes/Registrationdetails.svelte [hunk 5]
```diff
@@ -2,12 +2,214 @@
...
+    });
+  });
+  let state = {
+    name: '',
+    department: '',
+    batch: null,
+    phone: `+91`,
+    email: '',
+    password: '',
+    confirmpass: ''
```
Comment (AananthV) [inline src/routes/Registrationdetails.svelte:32]: Change to `confirmPassword`

File: src/routes/Registrationdetails.svelte [hunk 6]
```diff
@@ -2,12 +2,214 @@
...
+  };
+  function handlechange(e) {
```
Comment (AananthV) [inline src/routes/Registrationdetails.svelte:34]: use CamelCase for all function names and variables

File: src/routes/Register.svelte [hunk 7]
```diff
@@ -6,6 +6,74 @@
...
+        .then(response => {
+          navigate('/registerdetails', { replace: true });
+          toasts.add({
+            title: 'Success',
+            description: response.data.message,
+            duration: 10000, // 0 or negative to avoid auto-remove
+            placement: 'bottom-right',
+            type: 'success',
+            showProgress: true,
+            theme: localStorage.getItem('DAuth-theme')
```
Comment (AananthV) [inline src/routes/Register.svelte:43]: Weren't you using a context for this?

File: src/utils/auth.ts [hunk 8]
```diff
@@ -0,0 +1,2 @@
+import { writable } from 'svelte/store';
+export const auth = writable(localStorage.getItem('isDAuth'));
```
Comment (AananthV) [inline src/utils/auth.ts:2]: Not checking on every refresh?

File: src/routes/Register.svelte (hunk 7 above)
Comment (nishihere19) [reply src/routes/Register.svelte:43]: except app.svelte, will replace everywhere else....app.svelte gets loaded before themes context is set...

File: src/utils/auth.ts (hunk 8 above)
Comment (nishihere19) [reply src/utils/auth.ts:2]: as in? on refresh, will reload all variables and stores right? Upon Login and Logout, I'm updating store.

File: src/routes/Registrationdetails.svelte (hunk 4 above)
Comment (nishihere19) [reply src/routes/Registrationdetails.svelte:42]: yeah...will change

File: webpack.config.ts (hunk 1 above)
Comment (nishihere19) [reply webpack.config.ts:206]: I forgot to remove it...will do.

**Pull request merged**